src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -106,9 +129,17 @@ object Scope {
   }
 
   @tailrec
-  def fullyQualifiedNameBuilder(scope: Option[Scope], fqn: String, fullDisplay: Boolean): String = scope match {
-    case None => fqn.tail //Strip away the first "." of the name
-    case Some(x: Scope) => fullyQualifiedNameBuilder(x.parent, (if (fullDisplay || x.appearsInFQN) s".${x.name}" else "") + fqn, fullDisplay)
+  def fullyQualifiedNameBuilder(scope: Option[Scope], fqn: String, fullDisplay: Boolean, leaf: Boolean): String = {
+    scope match {
+      case None =>
+        fqn.tail //Strip away the first "." of the name
```

Comment:
I admit to being one of those who likes to see the base case first to make sure it actually exists.  :stuck_out_tongue: